internal/dag/policy.go

```diff
@@ -809,3 +812,49 @@ func loadBalancerRequestHashPolicies(lbp *contour_api_v1.LoadBalancerPolicy, val
 	}
 
 }
+
+func serviceCircuitBreakerPolicy(s *Service, cb *contour_api_v1alpha1.GlobalCircuitBreakerDefaults, m *metrics.Metrics) *Service {
+	if s == nil {
+		return nil
+	}
+
+	if s.MaxConnections == 0 && cb != nil {
+		s.MaxConnections = cb.MaxConnections
+	}
+
+	if s.MaxPendingRequests == 0 && cb != nil {
+		s.MaxPendingRequests = cb.MaxPendingRequests
+	}
+
+	if s.MaxRequests == 0 && cb != nil {
+		s.MaxRequests = cb.MaxRequests
+	}
+
+	if s.MaxRetries == 0 && cb != nil {
+		s.MaxRetries = cb.MaxRetries
+	}
```

Comment:
If the global circuit breaker isn't `nil`, but the fields are omitted (since they're optional), are we relying on the envoy cluster defaults?

Based on the envoy cluster configs here, https://github.com/projectcontour/contour/blob/main/internal/envoy/v3/cluster.go#L81-L90, I'm worried that any provided value will change those defaults since they're not passed as `nil`, but instead the default `uint32` 0

Reply:
good catch!

Reply:
Now I remember, darn I felt into this rabbit `hole` again

```
// UInt32OrNil returns a wrapped UInt32Value. If val is 0, nil is returned
func UInt32OrNil(val uint32) *wrapperspb.UInt32Value {
	switch val {
	case 0:
		return nil
	default:
		return wrapperspb.UInt32(val)
	}
}
```

handles 0 alright